generic type array can't clear() in a struct member function

### Describe the bug

When try to clear() a generic type array in a struct member function, the compiler generate a wrong code.

### Expected Behavior

v test.v should compile without error

### Current Behavior

v test.v will fail with error:

```sh
==================
/tmp/v_1000/test_array_clear.7621945871300060432.tmp.c:12509: warning: implicit declaration of function 'Array_main__EventHandler_T_string_clear'
tcc: error: undefined symbol 'Array_main__EventHandler_T_string_clear'
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

### Reproduction Steps

test.v
```v
module main

struct Registry[T] {
mut:
	events []EventHandler[T]
}

struct EventHandler[T] {
	name T
}


fn (mut r Registry[T]) test()
{
	r.events << EventHandler[string] {
		name: 'test'
	}
	r.events.clear()
}

fn main() {
	mut registry := &Registry[string]{
		events: []
	}
	 registry.test()
	//registry.events << EventHandler[string]{
	//	name: 'test'
	//}
	//registry.events.clear()
}
```

### Possible Solution

If move the code out of the struct member function, it will generate the correct code:
```v
module main

struct Registry[T] {
mut:
	events []EventHandler[T]
}

struct EventHandler[T] {
	name T
}

/*
fn (mut r Registry[T]) test()
{
	r.events << EventHandler[string] {
		name: 'test'
	}
	r.events.clear()
}
*/
fn main() {
	mut registry := &Registry[string]{
		events: []
	}
	 //registry.test()
	registry.events << EventHandler[string]{
		name: 'test'
	}
	registry.events.clear()
}
```

### Additional Information/Context

In struct  member function, it generate the following C code:
```c
VV_LOCAL_SYMBOL void main__Registry_T_string_test_T_string(main__Registry_T_string* r) {
        array_push((array*)&r->events, _MOV((main__EventHandler_T_string[]){ ((main__EventHandler_T_string){.name = _SLIT("test"),}) }));
        Array_main__EventHandler_T_string_clear(&r->events);
}

```

If move the code out of the struct member function, it will generate the correct code:
```c
        array_push((array*)&registry->events, _MOV((main__EventHandler_T_string[]){ ((main__EventHandler_T_string){.name = _SLIT("test"),}) }));
        array_clear(&registry->events);
```

### V version

V 0.3.4 01b2048

### Environment details (OS name and version, etc.)

V full version: V 0.3.4 01b2048
OS: linux, Ubuntu 22.04.2 LTS
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-9700 CPU @ 3.00GHz

getwd: /home/mars/v/8
vexe: /HD/github/kbkpbot/v_eventbus/v
vexe mtime: 2023-06-27 00:32:19

vroot: OK, value: /HD/github/kbkpbot/v_eventbus
VMODULES: OK, value: /home/mars/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.34.1
Git vroot status: weekly.2023.23-116-g02c38256-dirty
.git/config present: true

CC version: cc (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3-dirty

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -1281,7 +1281,7 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 		receiver_type_name = 'map'
 	}
 	if final_left_sym.kind == .array && !(left_sym.kind == .alias && left_sym.has_method(node.name))
-		&& node.name in ['repeat', 'sort_with_compare', 'free', 'push_many', 'trim', 'first', 'last', 'pop', 'clone', 'reverse', 'slice', 'pointers'] {
+		&& node.name in ['clear', 'repeat', 'sort_with_compare', 'free', 'push_many', 'trim', 'first', 'last', 'pop', 'clone', 'reverse', 'slice', 'pointers'] {
 		if !(left_sym.info is ast.Alias && typ_sym.has_method(node.name)) {
 			// `array_Xyz_clone` => `array_clone`
 			receiver_type_name = 'array'

--- a/vlib/v/tests/array_clear_test.v
+++ b/vlib/v/tests/array_clear_test.v
@@ -1,0 +1,29 @@
+struct Registry[T] {
+mut:
+	events []EventHandler[T]
+}
+
+struct EventHandler[T] {
+	name T
+}
+
+fn (mut r Registry[T]) test() {
+	mut events := []EventHandler[string]{}
+	r.events << EventHandler[string]{
+		name: 'test'
+	}
+	assert 1 == r.events.len
+
+	r.events.clear()
+	events.clear()
+
+	assert events.len == r.events.len
+}
+
+fn test_main() {
+	mut registry := &Registry[string]{
+		events: []
+	}
+	registry.test()
+	assert dump(registry.events).len == 0
+}

